eclass/kernel-build.eclass

```diff
@@ -436,6 +436,36 @@ kernel-build_merge_configs() {
 		fi
 	fi
 
+	if [[ ${KERNEL_IUSE_SECUREBOOT} ]]; then
+		if use secureboot; then
+			# This only effects arm64 and riscv where the bootable image may
+			# contain its own decompressor (zboot). If enabled, both the
+			# decompressor and the embedded kernel must be signed, the kernel
+			# build system can take care of this.
+			cat <<-EOF > "${WORKDIR}/secureboot.config" || die
+				## Enable zboot signing
+				CONFIG_EFI_ZBOOT=y
+				CONFIG_EFI_ZBOOT_SIGNING=y
```

Comment:
Doesn't that imply that we don't need the second signing logic in the eclass?

Reply:
Actually, I just grepped 6.4.13 and I don't see any `EFI_ZBOOT_SIGNING` in there.

Reply:
Now I am confused,

I found it in some documentation [here](https://cateee.net/lkddb/web-lkddb/EFI_ZBOOT_SIGNING.html) maybe this is not mainlined yet. I'll look a bit deeper into this.

In any case, with `CONFIG_EFI_ZBOOT` we should get a signable image. If it is not signed by the kernel then it will be signed by the eclass.

Reply:
Sure but I'd rather not add untested logic right now, so let's limit it to the first option.

Reply:
Looks like this was considered for 6.1, but ultimately got removed again:
https://lore.kernel.org/all/20221017171700.3736890-1-ardb@kernel.org/T/

According to this, we should be fine with just signing the image as a whole, signing the kernel inside the compressed image is not necessary any more.